java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -226,6 +226,14 @@ private void prune(Block inactiveBlock) {
         exitBlock = inactiveBlock.successors.iterator().next();
       }
     }
+
+    public boolean isRegularSuccessorOf(CFG.Block block) {
+      return this != block.exitBlock() || isMethodExitBlock();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)